test/RewardDelegators.js

```diff
@@ -182,6 +178,58 @@ contract("RewardDelegators contract", async function(accounts) {
         // update rewards when there are pending rewards for cluster and check if cluster is getting correct commission and also that accRewardPerShare is getting updated correctly
         // If weightedStake is 0, then check that no rewards are distributed
         // If rewards exist and then weightedStake becomes 0, then rewards still have to be distributed
+
+        // If weightedStake is 0, then check that no rewards are distributed
+        const weightedStake = await rewardDelegators.getClustersWeightedStake(registeredCluster);
+        assert.equal(Number(weightedStake), 0);
+
+        const clusterBeforeReward = await clusterRewards.clusterRewards(registeredCluster);
+        assert.equal(Number(clusterBeforeReward), 0);
+
+        await rewardDelegators._updateRewards(registeredCluster, { from: rewardDelegatorsOwner });
+
+        const clusterAfterReward = await clusterRewards.clusterRewards(registeredCluster);
+        assert.equal(Number(clusterAfterReward), 0);
+
+        // Check For Correct Update Case
+        const commission = 5;
+        await clusterRegistry.register(web3.utils.keccak256("DOT"), commission, registeredClusterRewardAddress, clientKey, {
+            from: registeredCluster
+        });
+
+        await delegate(delegator, [registeredCluster], [0], [2000000]); // failing
+        // await delegate(delegator, [registeredCluster], [1], [2000000]); // passing
+        await feedData([registeredCluster]);
+
+        const clusterUpdatedReward = await clusterRewards.clusterRewards(registeredCluster);
+        assert.equal(Number(clusterUpdatedReward), 3333);
+
+        const rewardAddrOldBalance = await PONDInstance.balanceOf(registeredClusterRewardAddress);
+        assert.equal(Number(rewardAddrOldBalance), 0);
+
+        const accPondRewardPerShareBefore = await rewardDelegators.getAccRewardPerShare(registeredCluster, PONDTokenId);
+        const accMPondRewardPerShareBefore = await rewardDelegators.getAccRewardPerShare(registeredCluster, MPONDTokenId);
+        assert.equal(Number(accPondRewardPerShareBefore), 0);
+        assert.equal(Number(accMPondRewardPerShareBefore), 0);
+
+        await rewardDelegators._updateRewards(registeredCluster, { from: rewardDelegatorsOwner });
+
+        // Checking Cluster Reward
+        const cluster1UpdatedRewardNew = await clusterRewards.clusterRewards(registeredCluster);
+        assert.equal(Number(cluster1UpdatedRewardNew), 0);
+
+        // Checking Cluster Commission
+        const rewardAddrNewBalance = await PONDInstance.balanceOf(registeredClusterRewardAddress);
+        assert(rewardAddrOldBalance != rewardAddrNewBalance);
+
+        // the actual rewardAddrNewBalance is 166.65 but due to solidity uint, it'll only be 166
+        assert.equal(Number(rewardAddrNewBalance), Math.floor(Number(clusterUpdatedReward) / 100 * commission));
+
```

Comment:
@prateekreddy the actual commission is 166.65 & the received commission is 166, i guess its fine.